telegram/ext/_basepersistence.py

```diff
@@ -398,50 +395,50 @@ def _insert_bot(self, obj: object, memo: Dict[int, object]) -> object:
         return obj
 
     @abstractmethod
-    def get_user_data(self) -> DefaultDict[int, UD]:
+    def get_user_data(self) -> Union[DefaultDict[int, UD], Dict[int, UD]]:
```

Comment:
why the union? `defaultdict` is a subclass of `dict`, so that should be clear anyway … same at a few other places

If there's no compelling reason to keep `defaultdict` in type hints & comments, I'd also suggest to add `.. versionchanged::` directives along the lines "May now return ordinary dicts instead of defaultdicts"

Reply:
I thought mypy would complain, since `dict` is not an instance of `defaultdict`. anyway will check with our current mypy version and latest version

Reply:
mh, where would it complain? i.e. where do we still use that `get_user_data()` is a `defaultdict`?